extensions-contrib/google-extensions/src/main/java/io/druid/storage/google/GoogleDataSegmentPusher.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.storage.google;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.InputStreamContent;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.metamx.common.CompressionUtils;
+import com.metamx.common.logger.Logger;
+import io.druid.segment.SegmentUtils;
+import io.druid.segment.loading.DataSegmentPusher;
+import io.druid.segment.loading.DataSegmentPusherUtil;
+import io.druid.timeline.DataSegment;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+public class GoogleDataSegmentPusher implements DataSegmentPusher
+{
+  private static final Logger LOG = new Logger(GoogleDataSegmentPusher.class);
+
+  private final GoogleStorage storage;
+  private final GoogleAccountConfig config;
+  private final ObjectMapper jsonMapper;
+
+  @Inject
+  public GoogleDataSegmentPusher(
+      final GoogleStorage storage,
+      final GoogleAccountConfig config,
+      final ObjectMapper jsonMapper
+  )
+  {
+    this.storage = storage;
+    this.config = config;
+    this.jsonMapper = jsonMapper;
+
+    LOG.info("Configured Google as deep storage");
+  }
+
+  @Deprecated
+  @Override
+  public String getPathForHadoop(String dataSource)
+  {
+    return String.format("gs://%s/%s", config.getBucket(), config.getPrefix());
+  }
+
+  @Override
+  public String getPathForHadoop()
+  {
+    return null;
```

Comment:
somehow that message is now deleted. 
New link - https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/druid-user/bd9dXKcJ8tg